staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/validator.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"fmt"
+	"strings"
+
+	celtypes "github.com/google/cel-go/common/types"
+	v1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ Validator = &CELValidator{}
+
+// CELValidator implements the Validator interface
+type CELValidator struct {
```

Comment:
Instead of exporting a `cel.CELValidator`, export a `cel.NewValidator() Validator` function that returns a unexported `validator` implementation of `Validator`?

This minimizes what export and also avoids the repeated `cel` term in `cel.CELValidator` type name.

Same for `CELEvaluatorCompiler` and any other interface implementing types.

Reply:
Got it! Done

Reply:
@maxsmythe this is the conversation I'm referring to in the other comment. It's possible I misunderstood the direction given though.

Reply:
Seems like you interpreted it correctly to me. IMO the interface is fine, since it's consistent with a lot of other packages (controller-runtime in particular).

Another option would be to just not export an interface (which is pretty common in Golang), but TBH that more just makes maintenance easier.

TL;DR SGTM either way.